testSuite/shared/src/test/scala/org/scalajs/testsuite/javalib/time/InstantTest.scala

```diff
@@ -0,0 +1,414 @@
+    assertEquals(980975L, someNegativeInstant.until(somePositiveInstant, DAYS))
+  }
+
+  @Test def compareTo(): Unit = {
+    for (i <- samples)
+      assertEquals(0, i.compareTo(i))
+
+    assert(Instant.MIN.compareTo(Instant.MAX) < 0)
+    assert(Instant.MIN.compareTo(Instant.EPOCH) < 0)
+    assert(Instant.MAX.compareTo(Instant.MIN) > 0)
+    assert(Instant.MAX.compareTo(Instant.EPOCH) > 0)
+  }
+
+  @Test def isAfter(): Unit = {
+    assertFalse(Instant.MIN.isAfter(Instant.MIN))
+    assertFalse(Instant.MIN.isAfter(Instant.MAX))
+    assertFalse(Instant.MIN.isAfter(Instant.EPOCH))
+    assertTrue(Instant.MAX.isAfter(Instant.MIN))
+    assertFalse(Instant.MAX.isAfter(Instant.MAX))
+    assertTrue(Instant.MAX.isAfter(Instant.EPOCH))
+  }
+
+  @Test def isBefore(): Unit = {
+    assertFalse(Instant.MIN.isBefore(Instant.MIN))
+    assertTrue(Instant.MIN.isBefore(Instant.MAX))
+    assertTrue(Instant.MIN.isBefore(Instant.EPOCH))
+    assertFalse(Instant.MAX.isBefore(Instant.MIN))
+    assertFalse(Instant.MAX.isBefore(Instant.MAX))
+    assertFalse(Instant.MAX.isBefore(Instant.EPOCH))
+  }
+
+  @Test def toEpochMilli(): Unit = {
+    assertEquals(0L, Instant.EPOCH.toEpochMilli)
+    assertEquals(928392983942L, somePositiveInstant.toEpochMilli)
+    assertEquals(-83827873286912L, someNegativeInstant.toEpochMilli)
+  }
+
+  @Test def toStringOutput(): Unit = {
+    assertEquals("1970-01-01T00:00:00Z", Instant.EPOCH.toString)
+    assertEquals("-1000000000-01-01T00:00:00Z", Instant.MIN.toString)
+    assertEquals("+1000000000-12-31T23:59:59.999999999Z", Instant.MAX.toString)
+    assertEquals("1999-06-03T06:56:23.942Z", somePositiveInstant.toString)
+    assertEquals("-0687-08-07T23:38:33.088936253Z", someNegativeInstant.toString)
+  }
+
+  @Test def now(): Unit = {
+    assertNotNull(Instant.now)
+  }
+
+  @Test def ofEpochSecond(): Unit = {
+    assertEquals(Instant.EPOCH, Instant.ofEpochSecond(0))
+    assertEquals(Instant.EPOCH, Instant.ofEpochSecond(0, 0))
+
+    assertEquals(Instant.MIN, Instant.ofEpochSecond(-31557014167219200L))
+    assertEquals(Instant.MIN, Instant.ofEpochSecond(-31557014167219200L, 0))
+    assertEquals(Instant.MAX, Instant.ofEpochSecond(31556889864403199L, 999999999))
+
+    expectThrows(classOf[DateTimeException], Instant.ofEpochSecond(-31557014167219200L, Long.MinValue))
+
+    val limits = Seq(-31557014167219200L, 31557014167219200L)
+    val invalidNanos = Seq(Long.MinValue, -1L, 1000000000L, Long.MaxValue)
+
+    val invalidPairs = limits.flatMap(l => invalidNanos.map(n => (l, n))).filter {
+      case (a, b) => ((a < 0) && (b < 0)) || ((a > 0) && (b > 0))
+    }
+    for ((s, n) <- invalidPairs)
+      expectThrows(classOf[DateTimeException], Instant.ofEpochSecond(s, n))
+  }
+
+  @Test def ofEpochMilli(): Unit = {
+    assertEquals(Instant.EPOCH, Instant.ofEpochMilli(0))
+  }
+
+  @Test def from(): Unit = {
+    for (i <- samples)
+      assertSame(i, Instant.from(i))
+  }
```

Comment:
You should add some tests that cover some `TemporalAccessor` that are not `Instant`.